### ISSUE TITLE
General Repo Module

### ISSUE THREAD

**Issue by robert4man** [ohioit/ansible-collection-github#5]
As a GitHub organization member, I want to create and manage GitHub repositories using Ansible. This should include all current functionality exposed via the GitHub 'Create an organization repository' API endpoint and should work with both GitHub Enterprise and GitHub Cloud Orgs.

References:
https://docs.github.com/en/rest/reference/repos#create-an-organization-repository
https://docs.ansible.com/ansible/latest/collections/community/general/github_repo_module.html#community-general-github-repo-manage-your-repositories-on-github

**Comment by ikogan** [comment]
The Github API exposes _a lot_ of functionality. Do we really need all of it out the gate? Perhaps I'm interpreting "manage GitHub repositories" and "all functionality" too broadly.

**Comment by robert4man** [comment]
Fair point. Poor wording. I specifically mean everything covered under create repo or /orgs/{org}/repos endpoint.
